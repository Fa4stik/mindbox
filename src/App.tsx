import React from 'react';
import './App.css';
import ColumnList from "./components/columnList/ColumnList";
import {useTypeSelector} from "./hooks/useTypeSelector";
import Item from "./components/item/Item";
import cover from './images/cover.png'
import {getTodos} from "./store/selectors/getTodos/getTodos";

function App() {
    const state = useTypeSelector(getTodos)

    return (
        <div className="App">
            <div className="w-full h-full px-12 py-4">
                <img src={cover}
                     alt="Cover"
                     className="w-full h-24 object-cover rounded-lg mb-4"/>
                <div className="flex">
                    {state.boards.map((board, indexB) => {
                        return <ColumnList board={board}
                                           key={indexB}
                        >
                            {board.todos.map((todo, indexT) => {
                                return <Item board={board} todo={todo} key={(indexB+1)*indexT}/>
                            })}
                        </ColumnList>
                    })}
                </div>
            </div>
        </div>
    );
}

export default App;