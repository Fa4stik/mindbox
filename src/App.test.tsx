import React from 'react';
import {fireEvent, screen, waitFor} from '@testing-library/react';
import App from './App';
import renderWithProvider from "./tests/helpers/renderWithProvider";

describe('TESTS', () => {
    test('Board render', () => {
        renderWithProvider(<App/>);
        const board = screen.getAllByTestId('column-list')[0];
        expect(board).toBeInTheDocument();
        expect(board).toMatchSnapshot();
    })

    test('Card render', () => {
        renderWithProvider(<App/>);
        const card = screen.getAllByTestId('item-card')[0];
        expect(card).toBeInTheDocument();
        expect(card).toMatchSnapshot();
    })

    test('Count board', () => {
        renderWithProvider(<App/>)
        const columnLists = screen.getAllByTestId('column-list');
        expect(columnLists.length).toBe(3);
    })

    test('Count children board', () => {
        renderWithProvider(<App/>)
        const firstBoardChildren = screen.getAllByTestId('column-list')[0]
            .querySelectorAll('[data-testid="item-card"]')
            .length
        expect(firstBoardChildren).toBe(2);
    })

    test('Check styles board', () => {
        renderWithProvider(<App/>)
        const board = screen.getAllByTestId('column-list')[0];
        expect(board).toHaveClass('mr-12 max-w-[300px]');
    })

    test('Check styles card', () => {
        renderWithProvider(<App/>)
        const card = screen.getAllByTestId('item-card')[0]
        expect(card).toHaveClass('bg-gray-400 shadow-lg')
    })

    test('Add card on board', () => {
        renderWithProvider(<App/>);
        const btnAddCard = screen.getAllByTestId('add-card')[0];
        const countCardStart = screen.getAllByTestId('item-card').length;
        expect(countCardStart).toBe(6);
        fireEvent.click(btnAddCard)
        fireEvent.click(btnAddCard)
        const countCardEnd = screen.getAllByTestId('item-card').length;
        expect(countCardEnd).toBe(8);
    })

    test('Add card on first board', () => {
        renderWithProvider(<App/>);
        const btnFirstBoardCard = screen.getAllByTestId('column-list')[0].querySelector('[data-testid="add-card"]')
        fireEvent.click(btnFirstBoardCard as Element);
        expect(screen.getAllByTestId('column-list')[0].childElementCount).toBe(3);
    })

    test('Delete card on first board', () => {
        renderWithProvider(<App/>);
        const delCard = screen.getAllByTestId('column-list')[0]
            .querySelectorAll('[data-testid="item-card"]')[0]
            .querySelector('[data-testid="delete-card"]')
        fireEvent.click(delCard as Element);
        const countElBoard = screen.getAllByTestId('column-list')[0]
            .querySelectorAll('[data-testid="item-card"]')
        expect(countElBoard.length).toBe(1);
    })

    test('Edit card on first board', () => {
        renderWithProvider(<App/>);
        const editCard = screen.getAllByTestId('column-list')[0]
            .querySelectorAll('[data-testid="item-card"]')[0]
            .querySelector('[data-testid="edit-card"]')
        fireEvent.click(editCard as Element);
        const textareaCard = screen.getAllByTestId('column-list')[0]
            .querySelectorAll('[data-testid="item-card"]')[0]
            .querySelector('[data-testid="textarea-card"]')
        fireEvent.input(textareaCard as Element, {
            target: {
                value: 'testValueX029'
            }
        })
        fireEvent.click(screen.getAllByTestId('column-list')[0])
        const textTest = screen.getByText(/testValueX029/i)
        expect(textTest).toBeInTheDocument();
    })
})