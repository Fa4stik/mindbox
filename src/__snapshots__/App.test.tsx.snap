// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TESTS Board render 1`] = `
<div
  class="max-w-[300px] h-full
 bg-gray-100
 shadow-md
 flex flex-col justify-center items-center
 mr-12 p-5 last:mr-0 rounded-lg"
  data-testid="column-list"
>
  <div
    class="h-full font-semibold"
  >
    <h1>
      all
    </h1>
  </div>
  <div
    class="flex flex-col py-4"
  >
    <div
      class="flex flex-wrap"
    >
      <div
        class="bg-gray-400 shadow-lg py-3 px-5 mr-2 mb-2 rounded-lg relative min-w-[250px] text-left"
        data-testid="item-card"
        draggable="true"
      >
        all1
        <img
          alt="Edit"
          class="absolute right-1 top-1 h-3"
          data-testid="edit-card"
          src="edit.png"
        />
        <img
          alt="Delete"
          class="absolute left-1 top-1 h-3"
          data-testid="delete-card"
          src="delete.png"
        />
      </div>
      <div
        class="bg-gray-400 shadow-lg py-3 px-5 mr-2 mb-2 rounded-lg relative min-w-[250px] text-left"
        data-testid="item-card"
        draggable="true"
      >
        all2
        <img
          alt="Edit"
          class="absolute right-1 top-1 h-3"
          data-testid="edit-card"
          src="edit.png"
        />
        <img
          alt="Delete"
          class="absolute left-1 top-1 h-3"
          data-testid="delete-card"
          src="delete.png"
        />
      </div>
    </div>
  </div>
  <button
    data-testid="add-card"
  >
    <img
      alt="Add"
      class="h-6"
      src="more.png"
    />
  </button>
</div>
`;

exports[`TESTS Card render 1`] = `
<div
  class="bg-gray-400 shadow-lg py-3 px-5 mr-2 mb-2 rounded-lg relative min-w-[250px] text-left"
  data-testid="item-card"
  draggable="true"
>
  all1
  <img
    alt="Edit"
    class="absolute right-1 top-1 h-3"
    data-testid="edit-card"
    src="edit.png"
  />
  <img
    alt="Delete"
    class="absolute left-1 top-1 h-3"
    data-testid="delete-card"
    src="delete.png"
  />
</div>
`;
